bin/install-theos

```diff
@@ -0,0 +1,598 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Pretty print
+special() {
+	printf "\e[0;34m==> \e[1;34mTheos Installer:\e[m %s\n" "$1"
+}
+
+update() {
+	printf "\n\e[0;36m==> \e[1;36m%s\e[m\n" "$1"
+}
+
+common() {
+	printf "\n\e[0;37m==> \e[1;37m%s\e[m\n" "$1"
+}
+
+error() {
+	printf "\e[0;31m==> \e[1;31m%s\e[m\n" "$1"
+}
+
+
+# Root is no bueno
+if [[ $EUID -eq 0 ]]; then
+	error "Theos should NOT be installed with or run as root (su/sudo)!"
+	error "  - Please re-run the installer as a non-root user."
+	exit 1
+fi
+
+
+# Common vars
+PLATFORM=$(uname)
+CSHELL="${SHELL##*/}"
+SHELL_ENV="unknown"
+if [[ "$CSHELL" == "sh" || "$CSHELL" == "bash" ]]; then
+	SHELL_ENV="$HOME/.profile"
+elif [[ "$CSHELL" == "zsh" ]]; then
+	SHELL_ENV="$HOME/.zshenv"
+fi
+# TODO: CI support (non-interactive)
+NONINTERACTIVE=0
+if [[ -z $CI ]]; then
+	NONINTERACTIVE=1
+fi
+
+
+# The work
+darwin() {
+	common "Platform: Darwin"
+
+	# Check for Xcode
+	XCODE="/Applications/Xcode.app/"
+	if ! [[ -d "$XCODE" && "$(ls -A "$XCODE")" ]]; then
+		error "Xcode is mandatory. The Command Line Tools package isn't sufficient for Theos to work. Xcode provides toolchains for all Apple platforms."
+		exit 3
+	fi
+
+	# Check for Homebrew
+	update "Preparing to install dependencies..."
+	if ! command -v brew &> /dev/null; then
+		read -p "Homebrew, which provides tools Theos depends on, is not installed. Would you like to have it installed for you? [y/n]" hbrew
+		if [[ "$hbrew" == "y" || "$hbrew" == "yes" || "$hbrew" == "Y" || "$hbrew" == "Yes" ]]; then
```

Comment:
Could we make a `bool` function for this? Like Theos’s `__theos_bool`.

Reply:
Ooo, good idea!